neo/Consensus/ConsensusService.cs

```diff
@@ -225,18 +273,268 @@ private void OnPersistCompleted(Block block)
         {
             Log($"persist block: {block.Hash}");
             block_received_time = TimeProvider.Current.UtcNow;
+            knownHashes.Clear();
             InitializeConsensus(0);
         }
 
+        private bool ReverifyPrepareRequest(ConsensusContext consensusContext, RecoveryMessage message, Snapshot snapshot,
+            out ConsensusPayload prepareRequestPayload, out PrepareRequest prepareRequest)
+        {
+            if (message.PrepareRequestMessage != null)
+            {
+                prepareRequest = message.PrepareRequestMessage;
+                prepareRequestPayload = consensusContext.RegenerateSignedPayload(
+                    prepareRequest, (ushort) consensusContext.PrimaryIndex,
+                    message.PreparationMessages[(ushort)consensusContext.PrimaryIndex].InvocationScript);
+
+                if (prepareRequestPayload.Verify(snapshot) && PerformBasicConsensusPayloadPreChecks(prepareRequestPayload))
```

Comment:
Can we reuse the snapshot in `ConsensusContext`?

https://github.com/neo-project/neo/blob/01f1f0e38f91077b265d5fb0005af50abd25020b/neo/Consensus/ConsensusContext.cs#L35

Reply:
Done.